apps/ewallet_db/lib/ewallet_db/user.ex

```diff
@@ -228,19 +236,19 @@ defmodule EWalletDB.User do
   """
   @spec insert(map()) :: {:ok, %User{}} | {:error, Ecto.Changeset.t()}
   def insert(attrs) do
-    multi =
-      Multi.new()
-      |> Multi.insert(:user, changeset(%User{}, attrs))
-      |> Multi.run(:wallet, fn %{user: user} ->
-        case User.admin?(user) do
+    %User{}
+    |> changeset(attrs)
+    |> Audit.insert(
```

Comment:
Maybe a future refactor suggestion...

The naming `Audit.insert()`, `Audit.update()`, etc. makes me think that it's the audit trail insert only, and not the user.

Could we make an Ecto facade, e.g. `AuditedEcto`, and change from `alias Ecto.Multi` to `alias EWalletDB.AuditedEcto.Multi` and so we could keep the same `Multi.insert()`?

Probably easier said than done 😅

Reply:
Hmmm, I don't really want to override an Ecto module with our own, makes it quite confusing I think 🤔 Maybe I'm not getting the idea tho. But I can rename the function to make it clearer.